src/parser/jsonld.ts

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+/* eslint-disable max-classes-per-file */
+import defaultFetch from "@inrupt/universal-fetch";
+import { promisifyEventEmitter } from "event-emitter-promisify";
+import type {
+  IJsonLdContext,
+  JsonLdContextNormalized,
+} from "jsonld-context-parser";
+import { ContextParser, FetchDocumentLoader } from "jsonld-context-parser";
+import { JsonLdParser } from "jsonld-streaming-parser";
+import { Store } from "n3";
+import CONTEXTS from "./contexts";
+import type { JsonLd } from "../common/common";
+
+/**
+ * A JSON-LD document loader with the standard context for VCs pre-loaded
+ */
+class CachedFetchDocumentLoader extends FetchDocumentLoader {
+  public async load(url: string): Promise<IJsonLdContext> {
+    if (Object.keys(CONTEXTS).includes(url)) {
+      return CONTEXTS[url as keyof typeof CONTEXTS];
+    }
+    // FIXME: See if we want to error on other contexts
+    return super.load(url);
+  }
+}
+
+/**
+ * Creates a context for use with the VC library
+ */
+// FIXME: See if our access grants specific context should be passed
+// through as a parameter instead
+export function getVcContext(
+  ...contexts: IJsonLdContext[]
+): Promise<JsonLdContextNormalized> {
+  const myParser = new ContextParser({
+    documentLoader: new CachedFetchDocumentLoader(),
+  });
+  return myParser.parse([...Object.values(CONTEXTS), ...contexts]);
+}
+
+interface Options {
+  fetch?: typeof globalThis.fetch;
+  baseIRI?: string;
+}
+
+/**
+ * Our internal JsonLd Parser with a cached VC context
+ */
+export class IJsonLdParser extends JsonLdParser {
```

Comment:
A capital `I` at the beginning of a symbol typically denotes an interface, which isn't the case here.